### ISSUE TITLE
Develop standards for tagging content

### ISSUE THREAD

**Issue by nikkelma** [hauler-dev/hauler#83]
Develop standards for managing tags in served registry.

Of particular interest:
- how should the same file be tagged and managed by the served registry?
- can multiple API objects add tags to the same repository target?

**Comment by joshrwolf** [comment]
This is a little tricky because of how legacy docker v1 handled implicit namespaces (`alpine:latest` --> `library/alpine:latest` _only_ on `docker.io`).

Ultimately want to abstract as much of this away from the user as possible, so the following proposed keeps the end ux at the forefront:

Adding/extracting unreferenced content to/from store
```bash
# add file to store
hauler store add file path/to/myfile.txt

# fetch file from store
hauler store extract myfile.txt

# add chart to store
hauler store chart path/to/my-chart

# fetch chart from store
hauler store x my-chart

# add image to store
hauler store add image alpine

# extract image from store
hauler store x library/alpine
```

Pushing/fetching unreferenced content from registries
```bash
REG="my-registry:5000"

# add file to store
hauler store add file path/to/myfile.txt

# add image to store
hauler store add image alpine

# add chart to store
hauler store add chart path/to/my-chart

# copy all store contents to registry
hauler store copy $REG

# download file from registry
hauler dl $REG/hauler/myfile.txt

# download image from registry
hauler dl $REG/library/alpine:latest

# download chart from registry
hauler dl $REG/hauler/my-chart
```

tldr:
* `download` commands _must_ be fully referenced content
* `extract` commands _can_ be shorthanded for files/charts, but images must include the full namespace
